doc/classes/CodeEdit.xml

```diff
@@ -601,6 +601,12 @@
 				Emitted when the user requests code completion.
 			</description>
 		</signal>
+		<signal name="new_line_inserted">
+			<param index="0" name="is_inserted_above" type="bool" />
+			<description>
+				Emitted when a new line has been inserted. As the text editor could have multiple carets, there is no information passed to the signal regarding the line position. Instead, use [method TextEdit.get_caret_count] to loop over all carets and [method TextEdit.get_caret_line] with the loop index to get the line position. If [param is_inserted_above] is [code]true[/code], the new line has been added above the caret.
```

Comment:
- A glimpse at the code tells me this signal is emitted once even with multiple carets, so it's actually "one or more lines"
- I feel like there's no need to explain ourselves here. Not too much, at least.
- A bit ambiguous. Does the example allow one to know the position of the caret's line or the newly created line? That I'm not sure of right now. This could actually use a code block example...
- Also could require a bit of shifting information around.
```suggestion
				Emitted when new lines have been inserted by a new line action, such as by pressing [kbd]Enter[/kbd]. If [param is_inserted_above] is [code]true[/code], the new lines have been added above the carets.
				Use [method TextEdit.get_caret_count] and [method TextEdit.get_caret_line] to infer the index of the newly created line.
```

Reply:
You prefer I use a callback to implement this feature and drop the signal style?

I'm not against it at all. I follow the recommendations and I want to be as clean as possible.

That's said, grazie for the review and the suggestion!

Reply:
> You prefer I use a callback to implement this feature and drop the signal style?

I'd **personally** prefer it, rather than adding something to the public API. 
Not saying the signal is useless, but it's out of the scope of this PR and could use a specific discussion about.

Reply:
I understand and it makes sens. I will implement it as soon as I have time 😊